src/main/java/org/apache/commons/dbcp2/BasicDataSource.java

```diff
@@ -221,6 +221,8 @@ protected static void validateConnectionFactory(final PoolableConnectionFactory
      */
     private boolean poolPreparedStatements = false;
 
+    private boolean clearStatementPoolOnReturn = false;
```

Comment:
Let's not continue the bad habit of initializing instance variables to their default values.

Reply:
Not sure what you mean here, Gary.  We definitely want this property to default to false.

Reply:
I get it now, but disagree.  The code is more readable IMO when the default is specified.

Reply:
I agree with Gary's other comments and added a couple of my own.  Nice work, Robert!  Nice thorough job updating everything.